XenAdmin/Controls/Ballooning/HostMemoryControls.cs

```diff
@@ -104,13 +98,25 @@ protected override void OnPaint(PaintEventArgs e)
 
             // Set the text values
             valueTotal.Text = Util.MemorySizeStringSuitableUnits(total, true);
-            valueUsed.Text = Util.MemorySizeStringSuitableUnits(used, true);
+            UpdateMemoryByMetricUpdater();
             valueAvail.Text = Util.MemorySizeStringSuitableUnits(avail, true);
             valueTotDynMax.Text = Util.MemorySizeStringSuitableUnits(tot_dyn_max, true);
             labelOvercommit.Text = string.Format(Messages.OVERCOMMIT, overcommit);
             valueControlDomain.Text = Util.MemorySizeStringSuitableUnits(dom0, true);
         }
 
+        public MetricUpdater MetricUpdater { get; set; }
+
+        public void UpdateMemoryByMetricUpdater()
+        {
+            if (MetricUpdater == null) return;
+            MetricUpdater.Prod();

```

Comment:
Calling `Prod` here causes another immediate update from the MetricUpdater, resulting in this page being continuously refreshed. This will happen on every paint event and on every other PropertyChanged event that we handle on this page. Also, because the `UpdateMemoryByMetricUpdater` is called from the MetricsUpdated event handler itself, it will again cause an immediate update of the metrics.